src/apps/backend/modules/task/rest-api/task-controller.ts

```diff
@@ -13,85 +13,81 @@ import {
+  createTask = applicationController(
+    async (req: Request<CreateTaskParams>, res: Response) => {
+      const task: Task = await TaskService.createTask({
+        accountId: req.accountId,
+        description: req.body.description,
+        title: req.body.title,
+      });
+      const taskJSON = serializeTaskAsJSON(task);
+      
+      res.status(HttpStatusCodes.CREATED).send(taskJSON);
+    },
+  );
 
-    res
-      .status(HttpStatusCodes.CREATED)
-      .send(taskJSON);
-  });
+  deleteTask = applicationController(
+    async (req: Request<DeleteTaskParams>, res: Response) => {
+      await TaskService.deleteTask({
+        accountId: req.accountId,
+        taskId: req.params.id,
+      });
 
-  deleteTask = applicationController(async (
-    req: Request<DeleteTaskParams>,
-    res: Response,
-  ) => {
-    await TaskService.deleteTask({
-      accountId: req.accountId,
-      taskId: req.params.id,
-    });
+      res.status(HttpStatusCodes.NO_CONTENT).send();
+    },
+  );
 
-    res
-      .status(HttpStatusCodes.NO_CONTENT)
-      .send();
-  });
+  getTask = applicationController(
+    async (req: Request<GetTaskParams>, res: Response) => {
+      const task = await TaskService.getTaskForAccount({
+        accountId: req.accountId,
+        taskId: req.params.id,
+      });
+      const taskJSON = serializeTaskAsJSON(task);
 
-  getTask = applicationController(async (
-    req: Request<GetTaskParams>,
-    res: Response,
-  ) => {
-    const task = await TaskService.getTaskForAccount({
-      accountId: req.accountId,
-      taskId: req.params.id,
-    });
-    const taskJSON = serializeTaskAsJSON(task);
-
-    res
-      .status(HttpStatusCodes.OK)
-      .send(taskJSON);
-  });
+      res.status(HttpStatusCodes.OK).send(taskJSON);
+    },
+  );
 
-  getTasks = applicationController(async (
-    req: Request,
-    res: Response,
-  ) => {
+  getTasks = applicationController(async (req: Request, res: Response) => {
     const page = +req.query.page;
     const size = +req.query.size;
+
+    // Convert sharedTask to a boolean
+    const sharedTask = req.query.sharedTask === 'true';
+
     const params: GetAllTaskParams = {
       accountId: req.accountId,
       page,
       size,
+      sharedTask, // Now sharedTask is correctly typed as a boolean
     };
+    let tasks: any[];
+    if (sharedTask === false) {
+      tasks = await TaskService.getTasksForAccount(params);
+    } else {
+      tasks = await TaskService.getSharedTasksForAccount(params); 
+    }
+    if(tasks.length === 0) {
+      res.status(HttpStatusCodes.OK).send([]);
+    }
 
-    const tasks = await TaskService.getTasksForAccount(params);
-    const tasksJSON = tasks.map((task) => serializeTaskAsJSON(task));
-
-    res
-      .status(HttpStatusCodes.OK)
-      .send(tasksJSON);
+    else{
+      const tasksJSON = tasks.map((task) => serializeTaskAsJSON(task)); 
+    res.status(HttpStatusCodes.OK).send(tasksJSON);
+    }
```

Comment:
You don't need to check the length of the task here.
`const tasksJSON = tasks.map((task) => serializeTaskAsJSON(task));` This is sufficient.